### ISSUE TITLE
Negative sampling 

### ISSUE THREAD

**Issue by TropComplique** [meereeum/lda2vec-tf#5]
The loss in your implementation is not exactly the same as in the original paper by Moody.
Am I right or wrong?

**Comment by meereeum** [comment]
@TropComplique you are correct! See relevant discussion [here](https://twitter.com/meereve/status/833096739666141185) - tldr; this is NCE, whereas Moody uses NS.

to quote the v helpful tf [docs](https://www.tensorflow.org/extras/candidate_sampling.pdf): "Negative sampling is a simplified variant of Noise Contrastive Estimation where we neglect to subtract off $log Q(y|x)$ during training."

in other words, NCE is not as sensitive to the choice of the sampling distribution over words - but is more computationally expensive than NS, which seems to work "well enough" for NLP corpora.

If you wanted to implement Negative Sampling (rather than Noise Contrastive Estimation), check out [`tf.nn.sampled_softmax_loss`](https://www.tensorflow.org/api_docs/python/tf/nn/sampled_softmax_loss)
